### PR TITLE
Fature/validation

### PR DESCRIPTION
**Pull request by franciscoPonceDev**
### Project requirements
- Add the following validations:
  - For the User model:
    - Name must not be blank.
    - PostsCounter must be an integer greater than or equal to zero.
  - For the Post model:
    - Title must not be blank.
    - Title must not exceed 250 characters.
    - CommentsCounter must be an integer greater than or equal to zero.
    - LikesCounter must be an integer greater than or equal to zero.
- Add unit specs for all of your models' methods and validations.
- Add flash messages in the create actions in all your controllers.
- Make sure that the N+1 problem is solved when fetching all posts and their comments for a user.

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -4,10 +4,11 @@ def create
     @comment = @post.comments.new(comment_params)
     @comment.user = current_user
     if @comment.save
-      redirect_to user_post_path(current_user, @post)
+      flash[:success] = 'Comment created!'
     else
-      render :new
+      flash[:error] = 'Comment not created!'
     end
+    redirect_to user_post_path(current_user, @post)
```
Comment (Deepakdanger) [inline app/controllers/comments_controller.rb:11]: - [ ] While creating a comment, I got this authentication error. 

Screenshot from 2022-06-10 23-22-53

Please read [this](https://stackoverflow.com/questions/35181340/rails-cant-verify-csrf-token-authenticity-when-making-a-post-request) to know more and fix this for now.

File: app/controllers/likes_controller.rb [hunk 2]
```diff
@@ -4,8 +4,13 @@ def create
...
 
-    @like = Like.create(user_id: current_user.id, post_id: @post.id)
+    @like = Like.new(user_id: current_user.id, post_id: @post.id)
 
+    if @like.save
+      flash[:success] = 'Liked!'
+    else
+      flash[:error] = 'ERROR!'
+    end
     redirect_to user_post_path(current_user, @post)
```
Comment (Deepakdanger) [inline app/controllers/likes_controller.rb:14]: - [ ] While creating a like, I got this authentication error. 

Screenshot from 2022-06-10 23-21-35


Please read [this](https://stackoverflow.com/questions/35181340/rails-cant-verify-csrf-token-authenticity-when-making-a-post-request) to know more and fix this for now.

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @franciscoPonceDev, 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Test is passing. :+1: 

## Required Changes ♻️
_Check the comments under the review._

- [ ] Also in the README please mention how to test the application. [Hinr: bundle exec rspec].

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by keneogu — APPROVED** [review]
> ## Hi @franciscoPonceDev,
> Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
> 
> ## Congratulations!     
> Cheers and Happy coding!clapclapclap
> 
> image [ ](https://camo.githubusercontent.com/2366be437d35bce33dd123620530e9131a8ac147fe960e76a36614dc703e8e22/68747470733a2f2f692e70696e696d672e636f6d2f6f726967696e616c732f37372f30652f62302f37373065623035373434393538326334343637326232643264316363333962362e676966) image [ ](https://camo.githubusercontent.com/2366be437d35bce33dd123620530e9131a8ac147fe960e76a36614dc703e8e22/68747470733a2f2f692e70696e696d672e636f6d2f6f726967696e616c732f37372f30652f62302f37373065623035373434393538326334343637326232643264316363333962362e676966)
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**